### ISSUE TITLE
Mongo not returning nested arrays. 

### ISSUE THREAD

**Issue by MilosStanic** [typeorm/typeorm#1766]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[x] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `websql`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I am using TypeORM with `NestJS`. 
I'm querying the the db as follows: 
```ts
async find(trazena_rec: string): Promise<RecnikActualId> {
    return await this.RecnikActualRepository.find( {
      where: { rec: trazena_rec }
    });
  }
```

The document I'm seeking is stored in DB as: 
```json
{
	"_id": ObjectId('5a1c51956fd7a9ca496237ce'),
	"na_slovo": "З",
	"rec": "заврнути",
	"ostatak_reci": ", заврнем сврш.",
	"izgovor": [
		"заврнути"
	],
	"definicije": [
		{
			"oznaka": "1.",
			"subDef": [
				{
					"oznaka": "а.",
					"sadrzaj": "окренути;окрећући затворити, прекинути, смањити,притегнути: ",
					"primer": []
				},
				{
					"oznaka": "б.",
					"sadrzaj": "затеорити; запречити. ",
					"primer": [
						{
							"sadrzaj": "— Били су душманиради тога што поп заврну сеоску благајну. ",
							"izvor": "Ђип. "
						},
						{
							"sadrzaj": "Јер су турски вардали улаци, да суВласи заврнули кланце. ",
							"izvor": "Март. '"
						}
					]
				}
			]
		},
		{
			"oznaka": "2.",
			"sadrzaj": "посувра-тити, подврнути, пресавивши мало задићи,загрнути. ",
			"primer": [
				{
					"sadrzaj": "— Заврни, сине Мрђане, заврними десну ногавицу. ",
					"izvor": "Кул. "
				},
				{
					"sadrzaj": "Пак заврну одзлата мараму, и угледа свога господара. ",
					"izvor": "НП Вук. '"
				}
			]
		},
		{
			"oznaka": "3.",
			"subDef": [
				{
					"oznaka": "а.",
					"sadrzaj": "скренути у страну, заеити. ",
					"primer": [
						{
							"sadrzaj": "— Затим су се освртали једно за другим,ј арочито на углу кад које требаше да заврнеУ другу улицу. ",
							"izvor": "Срем. "
						}
					]
				},
				{
					"oznaka": "б.",
					"sadrzaj": "дотаћи сечега као узгред, мало заћи, завирнути. —Успио је Пескаровић тек сасвим при кон-цу да некако заврне у питање. ",
					"primer": []
				}
			]
		},
		{
			"oznaka": "4.",
			"sadrzaj": "захватити, потерати, заератити1 (4). ",
			"primer": [
				{
					"sadrzaj": "— Се-дам-осам глава окинули и велики плијензаврнуше. ",
					"izvor": "НН Вук. '"
				}
			]
		},
		{
			"oznaka": "5.",
			"sadrzaj": "одвратити, одгоео-рити. ",
			"primer": [
				{
					"sadrzaj": "— Ја му заврнух: »Учинимо штогод«. ",
					"izvor": "Кур. '"
				}
			]
		},
		{
			"oznaka": "6.",
			"sadrzaj": "увртањем ушкопити. ",
			"primer": [
				{
					"sadrzaj": "— Мораћемога уштројити! — Заврнућемо му, као бику. ",
					"izvor": "Моск."
				}
			]
		}
	],
	"izvor": 27553
}
```

And what I'm getting back from TypeORM is this: 
```json
[
  {
      "_id":"5a1c51956fd7a9ca496237ce",
      "na_slovo":"З",
      "rec":"заврнути",
      "ostatak_reci":", заврнем сврш.",
      "izgovor":[
          "заврнути"
       ],
       "izvor":27553,       
      "definicije": [
        {
        "oznaka": "1.",
        "sadrzaj": undefined,
        "primer": undefined,
        "subDef": undefined },
       {
        "oznaka": "2.",
        "sadrzaj": "посувра-тити, подврнути, пресавивши мало задићи,загрнути. ",
        "primer": undefined,
        "subDef": undefined },
       {
        "oznaka": "3.",
        "sadrzaj": undefined,
        "primer": undefined,
        "subDef": undefined },
       {
        "oznaka": "4.",
        "sadrzaj": "захватити, потерати, заератити1 (4). ",
        "primer": undefined,
        "subDef": undefined },
       {
        "oznaka": "5.",
        "sadrzaj": "одвратити, одгоео-рити. ",
        "primer": undefined,
        "subDef": undefined },
       {
        "oznaka": "6.",
        "sadrzaj": "увртањем ушкопити. ",
        "primer": undefined,
        "subDef": undefined }
        ]
}
```

Please pay attention to the `definicije` array and its objects of which in the original document, some contain `subDef` arrays, and some contain `primer` arrays, none of which is included in the result returned from the server. 

Help, please. 

Thanks.

**Comment by sorooshansari** [comment]
Same problem here. any progress?

**Comment by Squallium** [comment]
I've just discovered the same problem in one of the latest version: 0.2.9-rc.1

**Comment by rustamwin** [comment]
Could you check latest version? Should be fixed in #3574
